samples/grids/grid/data-searching/src/index.tsx

```diff
@@ -35,9 +28,12 @@ const data = new MarketData();
 
 export default function Sample() {
   const gridRef = useRef<IgrGrid>(null);
+  const searchIconRef = useRef<IgrIconButton>(null);
```

Comment:
Do you need those refs? I think that you are not using them. Also, I think that we can remove all the refs and use events and state to handle the logic?